### ISSUE TITLE
"spack clean" does not work

### ISSUE THREAD

**Issue by eschnett** [LLNL/spack#1474]
A package (`julia`) failed to build. I want to re-build from scratch, and say `spack clean julia`. This does not seem to do anything; the next `spack install julia` immediately stops with the same error message. The source and build directories are not deleted, and `make` resumes where it stopped earlier.

I expect `spack clean` to delete the left-over build directory.

**Comment by tgamblin** [comment]
I'm not able to reproduce this.  On the latest develop:

```bash
(atala):spack$ spack stage julia
==> Trying to fetch from file:///Users/gamblin2/src/spack/var/spack/cache/julia/julia-0.4.6.tar.gz
curl: (37) Couldn't open file /Users/gamblin2/src/spack/var/spack/cache/julia/julia-0.4.6.tar.gz
==> Fetching from file:///Users/gamblin2/src/spack/var/spack/cache/julia/julia-0.4.6.tar.gz failed.
==> Trying to fetch from file:///Users/gamblin2/local/julia/julia-0.4.6.tar.gz
curl: (37) Couldn't open file /Users/gamblin2/local/julia/julia-0.4.6.tar.gz
==> Fetching from file:///Users/gamblin2/local/julia/julia-0.4.6.tar.gz failed.
==> Trying to fetch from https://github.com/JuliaLang/julia/releases/download/v0.4.6/julia-0.4.6-full.tar.gz
######################################################################## 100.0%
==> Staging archive: /Users/gamblin2/src/spack/var/spack/stage/julia-0.4.6-beku3gtrpzxslyx7kok545toaykur3jq/julia-0.4.6-full.tar.gz
==> Created stage in /Users/gamblin2/src/spack/var/spack/stage/julia-0.4.6-beku3gtrpzxslyx7kok545toaykur3jq
(atala):spack$ ls /Users/gamblin2/src/spack/var/spack/stage/julia-0.4.6-beku3gtrpzxslyx7kok545toaykur3jq/
julia-0.4.6/  julia-0.4.6-full.tar.gz
(atala):spack$ spack clean julia
(atala):spack$ ls /Users/gamblin2/src/spack/var/spack/stage/julia-0.4.6-beku3gtrpzxslyx7kok545toaykur3jq/
ls: cannot access '/Users/gamblin2/src/spack/var/spack/stage/julia-0.4.6-beku3gtrpzxslyx7kok545toaykur3jq/': No such file or directory
```

**Comment by tgamblin** [comment]
@eschnett: ping

**Comment by eschnett** [comment]
I managed to avoid the error that prevented Julia from being installed by updating curl. The system had a newer version of curl installed, and the way Spack interposes header files in the compiler flag led to the build problem -- curl would see the newer header files instead of its own. (We already discussed this in another issue, and the solution proposed there would have prevented this problem.)

At the moment I'm fighting a `spack reindex` problem again...

(Note to self: For the future I am not going to modify a set of installed packages any more. Instead, I am going to install Spack anew, into a new directory, and then build all packages anew. Instead of updating Spack and the packages in place, I'll reinstall everything. This has the advantage that existing packages won't break.)
